File: perl/singular_to_plural.pl
print("Please enter the singular word:\n");
my $word = <STDIN>;
chomp($word);

if ($word =~ /[s|sh|ch|o|x]$/) {
  print("${word}es\n");
  exit();
}
if ($word =~ /(.*)(fe?)$/) {
  print("${1}ves\n");
  exit();
}
if ($word =~ /(.*[^aiueo])y$/) {
  print("${1}ies\n");
  exit();
}

print("${word}s\n");

File: php/singular_to_plural.php
<?php
print('Please enter the singular word:' . PHP_EOL);
$word = trim(fgets(STDIN));

if (preg_match('/[s|sh|ch|o|x]$/', $word)) {
    print("{$word}es" . PHP_EOL);
    exit();
}
if (preg_match('/(.*)(fe?)$/', $word, $m)) {
    print("{$m[1]}ves" . PHP_EOL);
    exit();
}
if (preg_match('/(.*[^aiueo])y$/', $word, $m)) {
    print("{$m[1]}ies" . PHP_EOL);
    exit();
}

print("{$word}s" . PHP_EOL);
